### PR TITLE
Added operationName field to ClientMessage

### LINKED ISSUE #395: Missing operation name in ClientMessage
Add operation name to client message as an attribute as done in Java.

### PR DESCRIPTION
**Pull request by RikeVoltz**
This PR closes #395.

Changes in generated sources were made via some regex magic, probably I should look towards the possibility of implementing feature in [the generator](https://github.com/hazelcast/hazelcast-client-protocol) itself?

### CODE REVIEW

File: hazelcast/generated-sources/include/hazelcast/client/protocol/codec/ProtocolCodecs.h [hunk 1]
```diff
@@ -526,6 +526,7 @@ namespace hazelcast {
                 public:
                     static const ClientMessageType REQUEST_TYPE;
                     static const bool RETRYABLE;
+                    static const char* OPERATION_NAME;
```
Comment (ihsandemir) [inline hazelcast/generated-sources/include/hazelcast/client/protocol/codec/ProtocolCodecs.h:529]: better to use constexpr and initialize in place:
`static constexpr const char * OPERATION_NAME  =  "Client.AddMembershipListener";`

**Comment by ihsandemir** [comment]
> This PR closes #395.
> 
> Changes in generated sources were made via some regex magic, probably I should look towards the possibility of implementing feature in [the generator](https://github.com/hazelcast/hazelcast-client-protocol) itself?

Yes, we use https://github.com/hazelcast/hazelcast-client-protocol for generation. Current master of this repo uses 4.0 protocol version 2.0 which is different from existing in C++ which will updated soon for 4.0 release. You can look at https://github.com/hazelcast/hazelcast-client-protocol/tree/v1.8.0

Since the generator will be changed to implement the protocol version 2.0 codec for C++, for the time being, this manual changes may be OK, but how can we review these many changes? We need to say it is OK if the tests pass I guess, since they touch to all tests.

**Comment by ihsandemir** [comment]
verify

**Comment by RikeVoltz** [comment]
> Can you please add a test for using these new APIs for code coverage?

You mean small test for `.getOperationName()==.setOperationName()` or something more complicated?

File: hazelcast/test/src/HazelcastTests8.cpp [hunk 2]
```diff
@@ -2108,6 +2108,20 @@ namespace hazelcast {
     }
 }
 
+namespace hazelcast {
+    namespace client {
+        namespace test {
+            class ClientMessageTest: public ClientTestSupport {};
```
Comment (ihsandemir) [inline hazelcast/test/src/HazelcastTests8.cpp:2114]: PR looks good. Can you also add a test with an actual codec, e.g. `MapSizeCodec`? Encode a request and call getOperationName method.

**Comment by ihsandemir** [comment]
verify

**Comment by ihsandemir** [comment]
verify

**Comment by ihsandemir** [comment]
verify-linux

**Pull request merged**